### ISSUE TITLE
Base UDT not registered if first access isn't in/out/send/recv

### ISSUE THREAD

**Issue by jcflack** [tada/pljava#99]
Composite UDTs (and UDTs that mirror native PostgreSQL types) are registered by `sqlj.add_type_mapping`, but Base UDTs get registered in `Function.c` with the first call to one of their `input`/`output`/`send`/`receive` functions. That's all very well until a column of such a type is accessed in a way that doesn't call one of those first.

So, you can have:

```
# create table baz as select '(1,2)'::javatest.complex as z;
SELECT 1
postgres=# select javatest.logcomplex(z) from baz;
 logcomplex 
------------
 (1.0,2.0)
(1 row)
```

but start a clean session:

```
# \c
You are now connected to database "postgres"
postgres=# select javatest.logcomplex(z) from baz;
ERROR:  No java type mapping installed for "org.postgresql.pljava.example.annotation.ComplexScalar"
```

(Something like this should be added as a regression test some day when there is a better testing story; the need for a clean session rules out including it in the DDR.)

Two fix alternatives, neither one beautiful:

1. Declare that even base UDTs also need to be registered with `sqlj.add_type_mapping`. Almost works, but it turns out `UDT_registerUDT` takes an `isJavaBasedScalar` parameter, which is currently passed as always `false` by the code that finds entries in that map. Fixing *that* would require that code to examine the type much more closely to determine what kind of UDT it is, and that's roughly the same work as:
1. Add a test just before defaulting to `String` because no other mapping was found, and examine the type closely to see if it's a PL/Java UDT. Advantage: no need to fuss with `add_type_mapping` for base UDTs; disadvantage: it's a trail of many breadcrumbs from `pg_type` to a firm conclusion of PL/Java-UDT-ness.

**Comment by jcflack** [comment]
Addressed in 1.5.0-BETA3.

**Comment by jcflack** [comment]
Fix released in 1.5.0.
